### PR TITLE
QA-321: Add all components to the release tool

### PR DESCRIPTION
**Pull request by oleorhagen**
This includes support for:

* mender-binary-delta
* mender-convert
* mender-configure-module

Which are independent components, as well as `non-core`.

List them with `relase_tool --list --all`

Changelog: None
Signed-off-by: Ole Petter <ole.orhagen@northern.tech>

### CODE REVIEW

**Comment by oleorhagen** [comment]
Only tested meakly, by playing around with it locally.

Seems to work.

**Comment by oleorhagen** [comment]
```
$ release_tool --list --all
auditlogs
create-artifact-worker
deployments
deployments-enterprise
deviceadm
deviceauth
deviceauth-enterprise
deviceconfig
deviceconnect
devicemonitor
gui
integration
inventory
inventory-enterprise
iot-manager
mender
mender-api-gateway-docker
mender-artifact
mender-binary-delta <- New
mender-cli
mender-conductor
mender-conductor-enterprise
mender-configure-module <- New
mender-connect
mender-convert <- New
monitor-client
mtls-ambassador
reporting
tenantadm
useradm
useradm-enterprise
workflows
workflows-enterprise
```

File: component-maps.yml [hunk 1]
```diff
@@ -6,6 +6,27 @@
...
+
+  mender-binary-delta:
+    docker_image: []
+    docker_container: []
+    release_component: true
+    independent_component: true
+    non_core_component: true
+
+  mender-configure-module:
+    docker_image: []
```
Comment (lluiscampos) [inline component-maps.yml:24]: It is part of `mender-client-docker-addons` image

File: extra/release_tool.py [hunk 2]
```diff
@@ -172,6 +173,7 @@ def get_components_of_type(
             is_release_component = Component.COMPONENT_MAPS[type][comp][
                 "release_component"
             ]
+            non_core_component = Component.COMPONENT_MAPS.get(type,{}).get(comp, {}).get("non_core_component", False)
```
Comment (lluiscampos) [inline extra/release_tool.py:176]: Nitpick: for consistency with the others I suggest renaming to `is_non_core_component`

File: extra/release_tool.py [hunk 3]
```diff
@@ -153,6 +153,7 @@ def get_components_of_type(
         only_non_release=False,
         only_independent_component=False,
         only_non_independent_component=False,
+        only_core_components=True,
```
Comment (lluiscampos) [inline extra/release_tool.py:156]: By looking at all places we had to set `only_core_components=False` to call this, should we reconsider the default value?

**Review by lluiscampos — APPROVED** [review]
Looks good! Missing docker image for mender-configure-module and suggesting two internal things. Pre-aproving in any case.

Also:

> Only tested meakly, by playing around with it locally.

On the high level looks good: all parts for release and/or trigger builds use non-core components, while the behavior of cli options is not modified

> ```
> $ release_tool --list --all
> auditlogs
> ...
> deviceadm
> ...
> mender-binary-delta

While this is correct (`all` lists everything) I don't see it is much useful: I can either get core or get all (including deprecated repos!). Not required for this task, but maybe we want a cli option here to list "all active components" or "all Mender components" idk.

File: extra/release_tool.py (hunk 3 above)
Comment (oleorhagen) [reply extra/release_tool.py:156]: We could, I do like the explicitness though

Comment (lluiscampos) [reply extra/release_tool.py:156]: okey, no strong feelings here.

File: extra/release_tool.py (hunk 2 above)
Comment (oleorhagen) [reply extra/release_tool.py:176]: I'm probably misunderstanding, but the other's are named:
* `independent_component`
* `release_component`
Where does the `is` come in?

Comment (lluiscampos) [reply extra/release_tool.py:176]: I meant in the code, the Python variable. Two lines above you have `is_release_component = ...`.

**Pull request merged**